jetty-jaspi/src/main/config/modules/jaspi/jaspi-authmoduleconfig.xml

```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0"?>
+<!DOCTYPE Configure PUBLIC "-//Jetty//Configure//EN" "https://www.eclipse.org/jetty/configure_10_0.dtd">
+<Configure>
+  <!-- 
+       Example code to configure ServerAuthModules to be used by Jakarta Authentication 
+       using the programmatic Authentication API (registerConfigProvider)
+  -->
+  
+  <!-- 
```

Comment:
I think we should _not_ have commented out xml: it can't be verified and just rots.  So, my suggestion is that you move the `BasicAuthModule` up into `src/main/java/org/eclipse/jetty/security/jaspi/modules`  so it can actually be used.  Then, we can change the contents of this file to something like:

```suggestion
    <Ref refid="jaspiAuthConfigFactory">
	<Call name="registerConfigProvider">
          <Arg type="String">org.eclipse.jetty.security.jaspi.modules.BasicAuthModule</Arg>
          <Arg>
              <Map>
                  <Entry>
                      <Item>AppContextID</Item>
                      <Item>server /demo</Item>
                  </Entry>
                  <Entry>
                      <Item>org.eclipse.jetty.security.jaspi.modules.RealmName</Item>
                      <Item>TestRealm</Item>
                  </Entry>
              </Map>
          </Arg>
          <Arg type="String">HttpServlet</Arg>
          <Arg type="String">server /ctx</Arg>
          <Arg type="String">a test provider using BASIC authentication</Arg>
	</Call>
  </Ref>
</Configure>
```

We should put comments next to each arg/item to explain what they are.  Eg what is the arg `server /ctx`?? Is this a jaspi constant, or is it supposed to be the context path, or something else???

Reply:
method invoked is part of the AuthConfigFactory interface, which has javadoc., so not sure it needs anything else her?  e.g. "server /ctx" is example of a (logical) hostname + context path expalined in the spec.  "HttpServlet" is a constant in the spec (atm only profile supported here - the other, SOAP is deprecated and not supported here)

Reply:
I can't vouch for the quality of the Basic authentication auth module, but it looks Ok. Suggest this is marked as example

Reply:
Added some basic documentation and included the module setup above , but presumably most using Jakarta Authentication (JASPI) would be familiar with the specification.